File: editor/en/[2505]Bitwise OR of All Subsequence Sums.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def subsequenceSumOr(self, nums: List[int]) -> int:
        result = 0
        prefix_sum = 0

        for num in nums:
            prefix_sum += num
            result |= num
            result |= prefix_sum

        return result
# leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2505]Bitwise OR of All Subsequence Sums.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public long subsequenceSumOr(int[] nums) {
        long result = 0L;
        long sum = 0L;

        for (int num : nums) {
            sum += num;
            result |= num;
            result |= sum;
        }

        return result;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
